wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -30,6 +38,197 @@ sealed abstract class Wallet extends LockedWallet with UnlockedWalletApi {
     LockedWallet(nodeApi, chainQueryApi)
   }
 
+  /** We need a output so we can get proper tx size estimation */
+  private def getDummyDLCFundingOutputs(
+      amount: CurrencyUnit): Vector[TransactionOutput] = {
+    val key = ECPublicKey.freshPublicKey
+    val spk = P2WSHWitnessSPKV0(MultiSignatureScriptPubKey(2, Seq(key, key)))
+
+    // funded output and the other party's change
+    Vector(TransactionOutput(amount, spk), TransactionOutput(Satoshis.one, spk))
+  }
+
+  /** TODO Not currently used, will need to update wallet implementation for support  */
+  def writeDLCAddresses(eventId: String): Future[Vector[AddressDb]] = {
+    val addressDbs = {
+      val eventNum =
+        ByteVector.fromHex(eventId).get.take(4).toInt(signed = true).abs
+      val prefix = Vector(
+        BIP32Node(index = HDPurposes.SegWit.constant, hardened = true),
+        BIP32Node(index = HDCoinType.Testnet.toInt, hardened = true),
+        BIP32Node(index = eventNum, hardened = true)
+      )
+      val paths = Vector(
+        // Funding Key
+        SegWitHDPath(
+          prefix ++
+            Vector(BIP32Node(index = 0, hardened = false),
+                   BIP32Node(index = 0, hardened = false))).get,
+        // Refund Key
+        SegWitHDPath(
+          prefix ++
+            Vector(BIP32Node(index = 1, hardened = false),
+                   BIP32Node(index = 0, hardened = false))).get,
+        // Contract Execution Transaction (CET) Keys
+        SegWitHDPath(
+          prefix ++
+            Vector(BIP32Node(index = 1, hardened = false),
+                   BIP32Node(index = 1, hardened = false))).get,
+        SegWitHDPath(
+          prefix ++
+            Vector(BIP32Node(index = 1, hardened = false),
+                   BIP32Node(index = 2, hardened = false))).get,
+        // Closing Keys
+        SegWitHDPath(
+          prefix ++
+            Vector(BIP32Node(index = 2, hardened = false),
+                   BIP32Node(index = 0, hardened = false))).get,
+        SegWitHDPath(
+          prefix ++
+            Vector(BIP32Node(index = 2, hardened = false),
+                   BIP32Node(index = 1, hardened = false))).get,
+        SegWitHDPath(
+          prefix ++
+            Vector(BIP32Node(index = 2, hardened = false),
+                   BIP32Node(index = 2, hardened = false))).get
+      )
+
+      paths.map { addrPath =>
+        val pubkey = keyManager.getRootXPub.deriveChildPubKey(addrPath) match {
+          case Failure(exception) => throw exception
+          case Success(value)     => value.key
+        }
+
+        AddressDbHelper
+          .getSegwitAddress(pubkey, addrPath, networkParameters)
+      }
+    }
+
+    addressDAO.createAll(addressDbs)
+  }
+
+  /** This is needed to fill in values that we do not have yet */
+  private val dummyPartialSig: PartialSignature =
```

Comment:
this should probably be moved to an `object PartialSignature` along with `fromHex` :)